notebooks/sliceAnalysis.ipynb

```diff
@@ -0,0 +1,199 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Подключение библиотек"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nibabel as nib\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os\n",
+    "\n",
+    "import numpy as np\n",
+    "import cv2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Превращаем КТ в набор слайсов"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CT_PATH_FILE = os.path.join(\"assets\", \"input.nii.gz\")\n",
+    "CT_SLICES_DIR = os.path.join(\"output\", \"slices\")\n",
+    "\n",
+    "if not os.path.exists(CT_SLICES_DIR):\n",
+    "  os.mkdir(CT_SLICES_DIR)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Figure size 640x480 with 0 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "nii_path = CT_PATH_FILE\n",
+    "\n",
+    "nii_img = nib.load(nii_path)\n",
+    "\n",
+    "nii_data = nii_img.get_fdata()\n",
+    "\n",
+    "output_dir = CT_SLICES_DIR\n",
+    "if not os.path.exists(output_dir):\n",
+    "    os.makedirs(output_dir)\n",
+    "\n",
+    "for i in range(nii_data.shape[2]):\n",
+    "    plt.imshow(nii_data[:, :, i], cmap=\"gray\")\n",
+    "    plt.axis(\"off\")\n",
+    "    plt.savefig(os.path.join(output_dir, f\"slice_{i}.png\"), bbox_inches=\"tight\", pad_inches=0)\n",
+    "    plt.clf()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Анализируем контуры на слайсе"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "INPUT_FILE = os.path.join(CT_SLICES_DIR, \"slice_164.png\")\n",
+    "OUTPUT_FILE = os.path.join(\"output\", \"contoured_slice.png\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Смотрим на разные tresholds"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
```

Comment:
Small question: can we use link to the png image here in order to see only source code in the file and to visualize png at once?